Default value for the config key `environment` in the java sdk (#10911)
getsentry/sentry-docs: docs/platforms/java/common/configuration/options.mdx

<ConfigKey name="environment">

> Sets the environment. This string is freeform and is set to `production` as default. A release can be associated with more than one environment to separate them in the UI (think `staging` vs `prod` or similar).

Reviewer: ```suggestion
Sets the environment. This string is freeform and is set to `production` by default. A release can be associated with more than one environment to separate them in the UI (think `staging` vs `prod` or similar).
```

> By default the SDK will try to read this value from the `SENTRY_ENVIRONMENT` environment variable (except for the browser SDK where this is not applicable) and if this value is not set, it use the default `production`.

Reviewer: ```suggestion
By default, the SDK will try to read this value from the `SENTRY_ENVIRONMENT` environment variable (except for the browser SDK where this isn't applicable). If this value isn't set, it will use the default `production`.
```
